downloader/src/main/kotlin/JavaBootstrapper.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.downloader
+
+import eu.hansolo.jdktools.ArchiveType
+import eu.hansolo.jdktools.Latest
+import eu.hansolo.jdktools.LibCType
+import eu.hansolo.jdktools.Match
+import eu.hansolo.jdktools.OperatingSystem
+import eu.hansolo.jdktools.PackageType
+import eu.hansolo.jdktools.TermOfSupport
+import eu.hansolo.jdktools.util.Helper
+
+import io.foojay.api.discoclient.DiscoClient
+import io.foojay.api.discoclient.pkg.Scope
+
+import java.io.File
+
+import kotlin.time.measureTime
+import kotlin.time.measureTimedValue
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.utils.common.safeMkdirs
+import org.ossreviewtoolkit.utils.common.unpack
+import org.ossreviewtoolkit.utils.ort.Environment
+import org.ossreviewtoolkit.utils.ort.downloadFile
+import org.ossreviewtoolkit.utils.ort.okHttpClient
+import org.ossreviewtoolkit.utils.ort.ortToolsDirectory
+
+object JavaBootstrapper {
+    private val discoClient by lazy { DiscoClient(Environment.ORT_USER_AGENT) }
+
+    /**
+     * Return the single top-level directory contained in this directory, if any, or return this directory otherwise.
+     */
+    private fun File.singleContainedDirectoryOrThis() =
+        walk().maxDepth(1).filter { it != this && it.isDirectory }.singleOrNull() ?: this
+
+    /**
+     * Install a JDK matching [distributionName] and [version] below [ortToolsDirectory] and return its directory on
+     * success, or an exception on failure.
+     */
+    fun installJdk(distributionName: String, version: String): Result<File> {
+        val versionResult = eu.hansolo.jdktools.versioning.Semver.fromText(version)
+        if (versionResult.error1 != null) return Result.failure(versionResult.error1)
+
+        val semVer = versionResult.semver1
+
+        logger.info { "Setting up JDK '$distributionName' in version '$semVer'..." }
+
+        val os = Helper.getOperatingSystem()
+        val arch = Helper.getArchitecture()
```

Comment:
maybe `operatingSytem` and `architecture` are more speaking, even though these short form are not uncommon.